### PR TITLE
Changed the Solr service to be a native systemd service

### PR DESCRIPTION
**Pull request by dzdimov**
We previously had the SOLR service file masked as a systemd file but invoking init.d service which in turn did not report his true state to systemd hence after crash the service would not restart by itself.

We have changed the ExecStart and ExecStop to point to /solr/bin/solr and not to the init.d file.
By doing this change the solr service will now be a native systemd service.

### CODE REVIEW

**Review by kikomanaskov333 — DISMISSED** [review]

**Review by bojmil-armedia — DISMISSED** [review]

File: vagrant/provisioning/roles/solr/templates/solr.service [hunk 1]
```diff
@@ -1,17 +1,12 @@
...
-RemainAfterExit=true
-ExecStart={{ root_folder }}/app/solr/bin/init.d/solr start
-ExecStop={{ root_folder }}/app/solr/bin/init.d/solr stop
-# Starting from CentOS 7.7 we can't specify the PIDfile for 
-# solr anymore; but systemd can find it from the process 
-# table anyway.
-# PIDFile={{ root_folder }}/app/solr/bin/solr-8983.pid
 User=root
 Group=root
+PIDFile=/opt/app/arkcase/app/solr/bin/solr-8983.pid
```
Comment (fabricetriboix) [inline vagrant/provisioning/roles/solr/templates/solr.service]: I guess we shouldn't use a hardcoded path here, but `{{ root_folder }}` instead?

File: vagrant/provisioning/roles/solr/templates/solr.service [hunk 2]
```diff
@@ -1,17 +1,12 @@
...
 [Service]
 Type=forking
-RemainAfterExit=true
-ExecStart={{ root_folder }}/app/solr/bin/init.d/solr start
-ExecStop={{ root_folder }}/app/solr/bin/init.d/solr stop
-# Starting from CentOS 7.7 we can't specify the PIDfile for 
-# solr anymore; but systemd can find it from the process 
-# table anyway.
-# PIDFile={{ root_folder }}/app/solr/bin/solr-8983.pid
 User=root
```
Comment (fabricetriboix) [inline vagrant/provisioning/roles/solr/templates/solr.service]: I think the user should `solr`, not `root`

**Review by fabricetriboix — CHANGES_REQUESTED** [review]

File: vagrant/provisioning/roles/solr/templates/solr.service (hunk 1 above)
Comment (david-oc-miller) [reply vagrant/provisioning/roles/solr/templates/solr.service]: definitely need {{ root_folder }} here.

File: vagrant/provisioning/roles/solr/templates/solr.service (hunk 2 above)
Comment (david-oc-miller) [reply vagrant/provisioning/roles/solr/templates/solr.service]: Somehow solr requires to start as root user, then it forks a child process as the user indicated in the solr.in.sh file.

Comment (dzdimov) [reply vagrant/provisioning/roles/solr/templates/solr.service]: Changed the user to solr as well

Comment (fabricetriboix) [reply vagrant/provisioning/roles/solr/templates/solr.service]: @david-oc-miller When I tested that with @dzdimov , I actually had to set the user to `solr` in the unit file. @dzdimov Have you tested it with the `solr` user?

Comment (david-oc-miller) [reply vagrant/provisioning/roles/solr/templates/solr.service]: Well, maybe the requirement to start as root was imposed by the /etc/init.d-style script.  Now that we call solr directly you probably do have to run as solr user.  This revised service unit is a huge improvement over the old one!

**Review by kikomanaskov333 — APPROVED** [review]

**Pull request merged**